libnmstate/appliers/linux_bridge.py

```diff
@@ -29,3 +38,51 @@ def set_bridge_ports_metadata(master_state, slave_state):
     ports = master_state.get("bridge", {}).get("port", [])
     port = next(filter(lambda n: n["name"] == slave_state["name"], ports), {})
     slave_state["_brport_options"] = port
+
+
+def generate_port_vlan_metadata(desired_state, current_state):
+    desired_bridges = _filter_interface_type(
+        desired_state.interfaces, InterfaceType.LINUX_BRIDGE
+    )
+    current_bridges = _filter_interface_type(
+        current_state.interfaces, InterfaceType.LINUX_BRIDGE
+    )
+    desired_port_vlan_status_per_bridge = _get_port_vlan_status(
+        desired_bridges
+    )
+    current_port_vlan_status_per_bridge = _get_port_vlan_status(
+        current_bridges
+    )
+
+    for brname, brstate in desired_bridges.items():
+        desired_port_vlan_status = desired_port_vlan_status_per_bridge.get(
+            brname, {}
+        )
+        current_port_vlan_status = current_port_vlan_status_per_bridge.get(
+            brname, {}
+        )
+        current_port_vlan_status.update(desired_port_vlan_status)
```

Comment:
When certain slave interface been removed from this bridge, this might be wrong.

Please check whether slave still valid in this bridge.

Reply:
The desired bridges are gathered from `_filter_interface_type(...)` which only report bridges whose state is `UP`.

Does that address your concern ?